### PR TITLE
Share syscall definitions

### PR DESCRIPTION
**Pull request by TimePath**
Provide gamelogic->engine syscall definitions for use in alternate gamelogics without requiring them to duplicate between updates

### CODE REVIEW

File: src/cgame/cg_api.cpp [hunk 1]
```diff
@@ -39,7 +39,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 int VM::VM_API_VERSION = CGAME_API_VERSION;
 
-static IPC::CommandBufferClient cmdBuffer("cgame");
+extern IPC::CommandBufferClient cmdBuffer;
```
Comment (Kangz) [inline src/cgame/cg_api.cpp]: Not a fan of having an extern declared in a cpp. We could have a .h for that.

File: src/sgame/sg_api.cpp [hunk 2]
```diff
@@ -34,7 +34,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "shared/VMMain.h"
 #include "shared/CommonProxies.h"
 
-static IPC::SharedMemory shmRegion;
+extern IPC::SharedMemory shmRegion;
```
Comment (Kangz) [inline src/sgame/sg_api.cpp]: ditto

**Comment by Kangz** [comment]
LGTM, except that I think the VM::HandleSyscall functions should be moved too.

File: src/cgame/cg_api.cpp (hunk 1 above)
Comment (TimePath) [inline src/cgame/cg_api.cpp]: We have a .h for that

**Pull request merged**